src/flint/FlintTypes.jl

```diff
@@ -6693,3 +6840,78 @@ macro fq_default_mpoly_do_op(f, R, a...)
     return res
 end
 
+###############################################################################
+#
+#   Docstrings for the systematically added types in this file
+#
+###############################################################################
+
+module DocstringInfo
+using Markdown
+
+struct RingInfo
+    ring_name :: String
+    latex :: String
+end
+Base.convert(::Type{RingInfo}, (name, latex)::Tuple{String, String}) =
+    RingInfo(name, '$'*latex*'$')
+
+base_rings = Dict{Symbol, RingInfo}(
+    :ZZ => ("ZZRing", "\\mathbb Z"),
+    :QQ => ("QQField", "\\mathbb Q"),
+    :ZZMod => ("ZZModRing", "\\mathbb Z/n\\mathbb Z"),
+    :zzMod => ("zzModRing", "\\mathbb Z/n\\mathbb Z"),
+    :Fq => ("FqField", "\\mathbb F_q"),
+    :Fp => ("FpField", "\\mathbb F_p"),
+    :fp => ("fpField", "\\mathbb F_p"),
+    :FqPolyRep => ("FqPolyRepField", "\\mathbb F_q"),
+    :fqPolyRep => ("fqPolyRepField", "\\mathbb F_q")
+)
+
+struct ConstructionInfo
+    abstract_type :: String
+    super_type :: String
+    description :: String
+    reference :: String
+end
+Base.convert(::Type{ConstructionInfo}, x::NTuple{4, String}) = ConstructionInfo(x...)
+
+constructions = Dict{Symbol, ConstructionInfo}(
+    :MatrixSpace => ("MatSpace", "Module", "A matrix space", "matrix_space"),
+    :Matrix => ("MatElem", "ModuleElem", "A matrix", "matrix(::Ring)"),
+    :PolyRing => ("PolyRing", "Ring", "The polynomial ring", "polynomial_ring(R, :x)"),
+    :PolyRingElem => ("PolyRingElem", "RingElem", "A polynomial", "polynomial_ring(R, :x)"),
+    :MPolyRing => ("MPolyRing", "Ring", "A multivariate polynomial ring", "polynomial_ring(R, :x, :y)"),
+    :MPolyRingElem => ("MPolyRingElem", "RingElem", "A multivariate polynomial", "polynomial_ring(R, :x, :y)"),
+)
+
+@doc md"""
+    docstring(base::Symbol, suffix::Symbol)
+
+Docstring for some constructions of rings.
+
+# Examples
+```julia
+@doc docstring(:ZZ, :PolyRing)
+ZZPolyRing
+
+@doc Markdown.MD(docstring(:zzMod, :MatrixSpace), md"Contains $n^{rc}" elements.")
+zzModMatrixSpace
+```
+"""
+function docstring(base::Symbol, suffix::Symbol)
+    name = String(base) * String(suffix)
+    (; ring_name, latex) = base_rings[base]
+    (; abstract_type, super_type, description, reference) = constructions[suffix]
```

Comment:
With plain tuples this would become
```suggestion
    ring_name, latex = base_rings[base]
    abstract_type, super_type, description, reference = constructions[suffix]
```